test/tests/functional/pbs_ralter.py

```diff
@@ -2842,3 +2842,42 @@ def test_restart_revert(self):
         self.server.expect(RESV, attrs, id=rid)
         wait = start - time.time()
         self.check_resv_running(rid, offset=wait)
+
+    def test_alter_degrade_reconfirm_standing(self):
+        """
+        Test that if a standing reservation is altered, degraded,
+        then reconfirmed, the reservation will use the original
+        select
+        """
+        duration = 60
+        offset = 60
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'reserve_retry_time': 2})
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, standing=True, select="2:ncpus=2")
+
+        confirmed = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, confirmed, id=rid)
+
+        self.server.alterresv(rid, {ATTR_l: "select=1:ncpus=2"})
+        self.server.expect(RESV, confirmed, id=rid)
+
+        self.server.status(RESV, id=rid)
+        resv_node = self.server.reservations[rid].get_vnodes()[0]
+
+        offline = {'state': 'offline'}
+        self.server.manager(MGR_CMD_SET, NODE, offline, id=resv_node)
+        degraded = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
+        self.server.expect(RESV, degraded, id=rid)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': True})
```

Comment:
Why run a cycle?  The server will run one when the retry_time comes up.

Reply:
i forgot to turn off scheduling before offlining the node. This will prevent reserve_retry_time being too short.